edgelet/iotedge-diagnostics/src/main.rs

```diff
@@ -0,0 +1,144 @@
+                "unix" => {
+                    let client =
+                        hyper::Client::builder().build::<_, hyper::Body>(UnixConnector::new());
+                    let uri = UnixUri::new(
+                        management_uri
+                            .to_uds_file_path()
+                            .expect("couldn't get file path from URI"),
+                        "/modules/?api-version=2018-06-28",
+                    );
+                    client.get(uri.into())
+                }
+
+                "http" => {
+                    let client = hyper::Client::new();
+                    let uri = management_uri
+                        .join("/modules/?api-version=2018-06-28")
+                        .expect("could not construct list-modules request URI");
+                    client.get(uri.to_string().parse().expect(
+                        "could not convert list-modules request URI from url::Url to hyper::Uri",
+                    ))
+                }
+
+                scheme => panic!(
+                    "unrecognized scheme {:?} in management URI {:?}",
+                    scheme, management_uri
+                ),
+            };
+
+            let f = list_modules_response
+                .then(|response| {
+                    let response = response.expect("could not execute list-modules request");
+                    assert_eq!(
+                        response.status(),
+                        hyper::StatusCode::OK,
+                        "list-modules request did not succeed"
+                    );
+                    response.into_body().concat2()
+                })
+                .then(|body| {
+                    let body = body.expect("could not execute list-modules request");
+                    let _: management::models::ModuleList = serde_json::from_slice(&*body)
+                        .expect("could not parse list-modules response");
+                    Ok::<_, ()>(())
+                });
+
+            runtime.block_on(f).unwrap();
+        }
+
+        ("local-time", _) => {
+            println!(
+                "{}",
+                std::time::SystemTime::now()
+                    .duration_since(std::time::UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs()
+            );
+        }
+
+        ("idle-module", Some(matches)) => {
+            let duration = matches.value_of("DURATION").expect("unreachable");
+            let duration: u64 = duration.parse().expect("could not parse duration");
+
+            std::thread::sleep(std::time::Duration::from_secs(duration));
+        }
+
+        ("resolve-module", Some(matches)) => {
+            let hostname = matches.value_of("HOSTNAME").expect("unreachable");
+
+            let _ = std::net::ToSocketAddrs::to_socket_addrs(&(hostname, 80))
+                .unwrap_or_else(|err| panic!("could not resolve {}: {}", hostname, err))
```

Comment:
Do we want to panic here or return an error/error code?

Reply:
There's not much benefit to manually doing a `std::process::exit(1)` when panic will do that already, and print a helpful stack trace.

Reply:
I'm just sensitive to dumping a "scary" stack trace to the user when the error could be common. It seems like it would be better to print an error message and exit?

Reply:
Will the customer see the stack trace?

Reply:
As discussed, I'll change some of these to `eprintln!()` + `std::process::exit(1)`.